text/0000-else-match.md

```diff
@@ -0,0 +1,245 @@
+- Feature Name: Else Match
+- Start Date: 2016-07-26
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Extend the `if` expression to accept an `else match` block.
+
+# Motivation
+[motivation]: #motivation
+
+This proposal is meant to reduce the verbosity of writing `if ... else { match ... } ` style
+statements.
+
+It also makes code similar to:
+
+```rust
+let flag = false;
+
+if foo() {
+    do_this();
+} else {
+    match bar() {
+        baz() => do_that(),
+        _ => flag = true
+    }
+}
+
+if flag {
+    do_something();
+}
+```
+
+simpler and more concise:
+
+```rust
+if foo() {
+    do_this();
+} else match bar() {
+    baz() => do_that()
+} else {
+    do_something();
+}
+```
+
+## Use-cases
+
+Though rare, this pattern does exist in several Rust projects.
+
+### [Servo](https://github.com/servo/servo)
+
+<https://github.com/servo/servo/blob/master/components/layout/model.rs#L557>
+
+Before:
+
+```rust
+/// Clamp the given size by the given `min` and `max` constraints.
+pub fn clamp(&self, other: Au) -> Au {
+    if other < self.min {
+        self.min
+    } else {
+        match self.max {
+            Some(max) if max < other => max,
+            _ => other
+        }
+    }
+}
+```
+
+After:
+
+```rust
+/// Clamp the given size by the given `min` and `max` constraints.
+pub fn clamp(&self, other: Au) -> Au {
+    if other < self.min {
+        self.min
+    } else match self.max {
+        Some(max) if max < other => max,
+        _ => other
+    }
+}
```

Comment:
No, I'm sorry, to me, that doesn't look good at all. If you really want to get rid of that additional level of indentation, just use an early return in the if branch (works for the next example as well).